code/modules/mob/living/carbon/carbon.dm

```diff
@@ -509,9 +509,10 @@
 
 /mob/living/carbon/update_stamina(extend_stam_crit = FALSE)
 	var/stam = getStaminaLoss()
-	if(stam >= DAMAGE_PRECISION && (maxHealth - stam) <= crit_threshold && !stat && !HAS_TRAIT(src, TRAIT_NOSTAMCRIT))
+	if(stam >= DAMAGE_PRECISION && (maxHealth - stam) <= crit_threshold && !HAS_TRAIT(src, TRAIT_NOSTAMCRIT))
 		if(!stat)
-			enter_stamcrit()
+			if(extend_stam_crit || !HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, STAMINA))
```

Comment:
I believe this should be an && as the original behaviour was that taking stamina damage while stam critted would extend the regeneration timer by the standard stam regeneration time instead of by the stam crit delay

Reply:
5e3612c